defaults/defaults_unix.go.in

```diff
@@ -21,19 +21,19 @@ package defaults
 const (
 	// DefaultRootDir is the default location used by containerd to store
 	// persistent data
-	DefaultRootDir = "/var/lib/containerd"
+	DefaultRootDir = "@CONFIG_DEFAULT_ROOT_DIR@"
 	// DefaultStateDir is the default location used by containerd to store
 	// transient data
-	DefaultStateDir = "/run/containerd"
+	DefaultStateDir = "@CONFIG_DEFAULT_STATE_DIR@"
 	// DefaultAddress is the default unix socket address
-	DefaultAddress = "/run/containerd/containerd.sock"
+	DefaultAddress = "@CONFIG_DEFAULT_ADDRESS@"
 	// DefaultDebugAddress is the default unix socket address for pprof data
-	DefaultDebugAddress = "/run/containerd/debug.sock"
+	DefaultDebugAddress = "@CONFIG_DEFAULT_DEBUG_ADDRESS@"
 	// DefaultFIFODir is the default location used by client-side cio library
 	// to store FIFOs.
-	DefaultFIFODir = "/run/containerd/fifo"
+	DefaultFIFODir = "@CONFIG_DEFAULT_FIFO_DIR@"
 	// DefaultRuntime is the default linux runtime
-	DefaultRuntime = "io.containerd.runc.v2"
+	DefaultRuntime = "@CONFIG_DEFAULT_RUNTIME@"
 	// DefaultConfigDir is the default location for config files.
-	DefaultConfigDir = "/etc/containerd"
+	DefaultConfigDir = "@CONFIG_DEFAULT_CONFIG_DIR@"
```

Comment:
As mentioned, our config defaults are optionally overridable, both at the command line and/or in the configuration toml. 

Thus, if these defaults are changed, then each instance of containerd that does not already override the defaults will become broken.

I like @dmcgowan 's idea about a templated config or you could just pass the parameters at the command line...
```
USAGE:
   containerd [global options] command [command options] [arguments...]
   --config value, -c value     path to the configuration file (default: "/etc/containerd/config.toml")
   --address value, -a value    address for containerd's GRPC server
   --root value                 containerd root directory
   --state value                containerd state directory
```